libraries/AP_Airspeed/AP_Airspeed_Health.cpp

```diff
@@ -43,15 +45,36 @@ void AP_Airspeed::check_sensor_ahrs_wind_max_failures(uint8_t i)
         }
         return;
     }
-    const float speed_diff = fabsf(state[i].airspeed-gps.ground_speed());
 
+    // check for airspeed consistent with wind and vehicle velocity using the EKF
+    uint32_t age_ms;
+    float innovation, innovationVariance;
+    if (AP::ahrs().airspeed_health_data(innovation, innovationVariance, age_ms) && age_ms < 1000 && is_positive(innovationVariance)) {
+        state[i].failures.test_ratio = fabsf(innovation) / safe_sqrt(innovationVariance);
+    } else {
+        state[i].failures.test_ratio = 0.0f;
+    }
+    bool data_is_inconsistent;
+    if ((AP_Airspeed::OptionsMask::USE_EKF_CONSISTENCY & _options) != 0) {
+        float gate_size = MAX(_wind_gate, 0.0f);
+        if (param[i].use == 0) {
+            // require a smaller inconsistency for a disabled sensor to be declared consistent
+            gate_size *= 0.7f;
+        }
+        data_is_inconsistent = state[i].failures.test_ratio > gate_size;
+    } else {
+        data_is_inconsistent = false;
```

Comment:
```suggestion
```

Reply:
Setting to false on line 57 means we can remove the else case.

Reply:
That's a matter of style

Reply:
The patch I pulled in from Tridge that enables the consistency check independent of using the wind max method and that implements your suggestion.